projects/storefrontlib/cms-components/misc/theme-switcher/theme-switcher.component.service.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { SiteThemeService, Theme } from '@spartacus/core';
+import { ThemeService } from '../../../layout/theme/theme.service';
+import { Observable, tap } from 'rxjs';
+
+@Injectable()
+export class ThemeSwitcherComponentService {
+  constructor(
+    protected themeService: ThemeService,
+    protected siteThemeService: SiteThemeService
+  ) {}
+
+  getItems(): Observable<Theme[]> {
+    return this.siteThemeService.getAll();
+  }
+
+  getActiveItem(): Observable<string> {
+    return this.siteThemeService
+      .getActive()
+      .pipe(tap((theme) => this.themeService.setTheme(theme)));
+  }
```

Comment:
getter of the component should not set the theme for the page.
Instead maybe the `ThemeService.init()` should use the state from ngrx state that you created?
anuyways, I'm confused that now we set the theme both in the APP_BOOSTRAP_LISTENER ->  ThemeService.init()`
AND
in this component

Reply:
see point (4) in the comment: https://github.com/SAP/spartacus/pull/19016#discussion_r1700429306